### PR TITLE
Test clean up

### PR DESCRIPTION
**Pull request by andorp**
* Unifying tests with tasty framework
 * Creating TestSet monad to be able to create test sets and run them in the interpreter
 * SQL implementation of Notifications
 * Removing dead code
 * Removing WebDriver tests
 * Removing themis test framework

### CODE REVIEW

File: src/Bead/Domain/RolePermission.hs [hunk 1]
```diff
@@ -55,9 +57,13 @@ admin P_Delete o = elem o [P_Course, P_CourseAdmin, P_GroupAdmin]
 
 #ifdef TEST
 
--- * Invariants
+permissionTest =
+  assertProperty
+    "Permission is a total function"
+    (\(r,p,o) -> length (show (permission r p o)) > 0)
+    (triplet <$> enumGen <*> enumGen <*> enumGen)
+    "Permission relation is totally defined"
```
Comment (pgj) [inline src/Bead/Domain/RolePermission.hs]: There should be a negation here (i.e. a missing "not")?

File: src/Bead/Domain/RolePermission.hs [hunk 2]
```diff
@@ -55,9 +57,13 @@ admin P_Delete o = elem o [P_Course, P_CourseAdmin, P_GroupAdmin]
...
+  where
+    triplet a b c = (a,b,c)
```
Comment (pgj) [inline src/Bead/Domain/RolePermission.hs]: `triplet` could have been written (simply) as `(,,)` (which is a built-in function)

File: src/Bead/Persistence/NoSQLDir.hs [hunk 3]
```diff
@@ -1314,6 +1314,6 @@ encodePwd :: String -> String
 encodePwd = ordEncode
 
 #ifdef TEST
```
Comment (pgj) [inline src/Bead/Persistence/NoSQLDir.hs]: This should be removed fully instead.

File: src/Bead/View/Pagelets.hs [hunk 4]
```diff
@@ -555,12 +555,13 @@ verticalRadioButtons name values = mapM_ button values
...
 
--- * Invariants
-
-invariants :: Invariants P.PageDesc
-invariants = Invariants [
-    ("Page link text must be defined: ", \p -> length (linkText' p) > 0)
-  ] where
+linkTextTest =
+  assertProperty
+    "Page link is atotal function"
```
Comment (pgj) [inline src/Bead/View/Pagelets.hs]: "atotal" should be "a total".

File: src/Test/Tasty/Arbitrary.hs [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+import Test.QuickCheck.Arbitrary
+import Test.QuickCheck.Gen
+
+alpha = elements ['a' .. 'z']
+num   = elements ['0' .. '9']
+
+alphaNum = oneof [alpha, num]
+
+class Interval n where
```
Comment (pgj) [inline src/Test/Tasty/Arbitrary.hs]: The introduction of the `Interval` class seems to be unnecessary as it is not exploited too much in the rest of the code.

File: test/Test/Unit/Module.hs [hunk 6]
```diff
@@ -0,0 +1,62 @@
+{-# LANGUAGE CPP #-}
+--{-# LANGUAGE TypeSynonymInstances #-}
```
Comment (pgj) [inline test/Test/Unit/Module.hs]: Potential cruft, these should be removed.

File: src/Bead/Persistence/NoSQLDir.hs (hunk 3 above)
Comment (andorp) [inline src/Bead/Persistence/NoSQLDir.hs]: This is a placeholder for test in different implementations for the persistence layer.

**Pull request merged**